backend/services/dynamodb_service.py

```
@@ -73,11 +73,32 @@ def get_scan_record(self, scan_id: str, timestamp: str) -> Optional[Dict]:
+                if limit is not None and len(items) >= scan_kwargs["Limit"]:
+                    break
+
+                scan_kwargs["ExclusiveStartKey"] = response["LastEvaluatedKey"]
+                response = self.scan_results.scan(**scan_kwargs)
+                items.extend(response.get('Items', []))
+
+            if limit is not None:
+                return items[: scan_kwargs["Limit"]]
+            return items
         except ClientError as e:
             logger.error(f"DynamoDB error listing scan records: {str(e)}")
```

Ruff (0.15.9)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)